sql/insert.go

```diff
@@ -88,6 +95,22 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 				b.Put(key, string(t))
```

Comment:
I think if you used `CPut` instead of `Put` here and required no existing value to be present (`expValue == nil`) then we would get the appropriate error on trying to insert an already existing value. Same idea could be used for other unique indexes.